### PR TITLE
chore: Use standard cmake for static/shared lib selection

### PR DESCRIPTION
**Pull request by robinlinden**
Instead of building both variants depending on how flags are set, usually you just use the CMake variable `BUILD_SHARED_LIBS` to select one or the other.

https://cmake.org/cmake/help/latest/variable/BUILD_SHARED_LIBS.html

---
This change is [Reviewable](https://reviewable.io/reviews/toktok/c-toxcore/1351)

### CODE REVIEW

**Comment by avoidr** [comment]
If this means that building shared and static library is mutually exclusive, then I can only ask you to reconsider this, as it makes things unnecessarily complicated for static linking users.

0. The project needs to be configured and installed twice, two separate times. That's an unreasonable waste of resources only to cat together a bunch of object files, which are created in any case, anyway.
1. Repositories of distributions will no longer provide the static libraries. Nobody is going to build a `lib*-static` package just for a single file, and can't be reasonably expected to.
2. Users who compile their application have to go from merely passing `-static` to the compiler, to fetching and building the whole library by themselves, again, only for one file.

Libraries are not the ones to decide how to link their library. That's the user's choice, and at best, the distro packager's.

**Comment by iphydf** [comment]
@robinlinden are @avoidr's concerns valid?

File: INSTALL.md [hunk 1]
```diff
@@ -110,7 +108,7 @@ Example of calling cmake with options
 
 ```sh
 cmake \
-  -D ENABLE_STATIC=OFF \
+  -D BUILD_SHARED_LIBS=OFF \
```
Comment (iphydf) [inline INSTALL.md]: I don't know the context here, but "ENABLE_STATIC=OFF" -> "BUILD_SHARED_LIBS=OFF" sounds odd.

**Comment by avoidr** [comment]
Here is another discussion on this topic: https://github.com/mesonbuild/meson/issues/484

**Comment by robinlinden** [comment]
This is a more standard way of dealing with static/shared libs in CMake (you can set BUILD_SHARED_LIBS for any library in CMake and it will produce shared libraries), but I don't care that much if we prefer the custom ModulePackage method of building things.

**Pull request closed without merging**